src/frontend/frontend/onboarding.py

```
@@ -0,0 +1,100 @@
+    needs = needs_admin_onboarding(context) if needs_onboarding is None else needs_onboarding
+    session[ADMIN_ONBOARDING_SESSION_KEY] = {
+        "needs_onboarding": needs,
+        "context": context if needs else None,
+    }
+    return context if needs else None
+
+
+def get_admin_onboarding_context() -> dict[str, bool | str] | None:
+    try:
+        settings = DataPersistenceLayer().get_first(Settings)
+    except Exception:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)